### PR TITLE
boards/iotlab: add DEBUG_ADAPTER_ID openocd

### PR DESCRIPTION
**Pull request by fjmolinas**
### Contribution description

If there are multiple ftdi device openocd will not be able to now which one is the `iotlab-m3` and will fail flashing. This can be ammended by setting a `DEBUG_ADAPTER_ID`, this is currently not passed for `iotlab-m3` boards, this PR ammends it.

**NOTE**: iotlab-m3 do not come with a serial number but what can easily right one to the ftdi eeprom (you will need to install `ftdi-eeprom`)

`ftdi_eeprom --flash-eeprom ~/workspace/ci-riot-tribe/udev/iotlab-m3.conf`

```
# Based on the examples in:
# https://waterpigs.co.uk/articles/ftdi-configure-mac-linux/

# The Vendor ID and Product ID seem to identify a particular chipset
# to target. I used the libftdi utilities and OS X USB Prober app to
# poke around and figure out what numbers to use.

vendor_id=0x403
product_id=0x6010

# The rest of the fields are settings which can be written to the
# FT2xx with the --flash-eeprom option.

# Max. power consumption: value * 2 mA. Use 0 if self_powered = true.
max_power=50

###########
# Strings #
###########
manufacturer="IoT-LAB"                  # Manufacturer
product="M3"                            # Product
serial="TRIBE4441125307157009"          # Serial


###########
# Options #
###########
self_powered=false  # Turn this off for bus powered
remote_wakeup=false # Turn this on for remote wakeup feature
use_serial=true     # Use the serial number string

```

### Testing procedure

I got this issue when having a `iotlab-m3` and openmote-b connected on the same usb-hub.

```
/sys/bus/usb/devices/1-1.5.3: IoT-LAB M3, serial: 'TRIBE4441125307150001', tty(s): ttyUSB3
/sys/bus/usb/devices/1-1.6: FTDI Dual RS232-HS, serial: '', tty(s): ttyUSB2

```

- I have an `opemote-b` and a `iotlab-m3`, so two ftdi devices used to flash. On master trying to glash `iotlab-m3` returns an error, it suceds with this PR

<details><summary>master</summary>

```
BOARD=iotlab-m3 make -C examples/hello-world/ all flash-onlu
/home/francisco/workspace/RIOT/dist/tools/openocd/openocd.sh flash /home/francisco/workspace/RIOT/examples/hello-world/bin/iotlab-m3/hello-world.elf
### Flashing Target ###
Open On-Chip Debugger 0.10.0+dev-00703-g92bb76a4-dirty (2019-07-19-14:27)
Licensed under GNU GPL v2
For bug reports, read
	http://openocd.org/doc/doxygen/bugs.html
Info : auto-selecting first available session transport "jtag". To override use 'transport select <transport>'.
adapter speed: 1000 kHz
adapter_nsrst_delay: 100
jtag_ntrst_delay: 100
none separate
cortex_m reset_config sysresetreq
trst_and_srst separate srst_nogate trst_push_pull srst_open_drain connect_deassert_srst
trst_and_srst separate srst_nogate trst_push_pull srst_open_drain connect_assert_srst
Info : clock speed 1000 kHz
Error: JTAG scan chain interrogation failed: all ones
Error: Check JTAG interface, timings, target power, etc.
Error: Trying to use configured scan chain anyway...
Error: stm32f1x.cpu: IR capture error; saw 0x0f not 0x01
Warn : Bypassing JTAG setup events due to errors
Error: Invalid ACK (7) in DAP response
Error: JTAG-DP STICKY ERROR
Error: Invalid ACK (7) in DAP response
Error: JTAG-DP STICKY ERROR
Error: Invalid ACK (7) in DAP response
Error: JTAG-DP STICKY ERROR
Error: Invalid ACK (7) in DAP response
Error: JTAG-DP STICKY ERROR
Error: Invalid ACK (7) in DAP response
Error: JTAG-DP STICKY ERROR
Error: Invalid ACK (7) in DAP response
Error: JTAG-DP STICKY ERROR
Error: Invalid ACK (7) in DAP response
Error: JTAG-DP STICKY ERROR
Error: Invalid ACK (7) in DAP response
Error: JTAG-DP STICKY ERROR
Error: Invalid ACK (7) in DAP response
Error: JTAG-DP STICKY ERROR
Error: Invalid ACK (7) in DAP response
Error: JTAG-DP STICKY ERROR
Error: Invalid ACK (7) in DAP response
Error: JTAG-DP STICKY ERROR
Error: Invalid ACK (7) in DAP response
Error: JTAG-DP STICKY ERROR
Error: Invalid ACK (7) in DAP response
Error: JTAG-DP STICKY ERROR
Error: Invalid ACK (7) in DAP response
Error: JTAG-DP STICKY ERROR
Error: Invalid ACK (7) in DAP response
Error: JTAG-DP STICKY ERROR
Error: Invalid ACK (7) in DAP response
Error: JTAG-DP STICKY ERROR
Error: Invalid ACK (7) in DAP response
Error: JTAG-DP STICKY ERROR
Error: Invalid ACK (7) in DAP response
Error: JTAG-DP STICKY ERROR
Error: Invalid ACK (7) in DAP response
Error: JTAG-DP STICKY ERROR
Error: Invalid ACK (7) in DAP response
Error: JTAG-DP STICKY ERROR
Error: Invalid ACK (7) in DAP response
Error: JTAG-DP STICKY ERROR
Error: Invalid ACK (7) in DAP response
Error: JTAG-DP STICKY ERROR
Error: Invalid ACK (7) in DAP response
Error: JTAG-DP STICKY ERROR
Error: Invalid ACK (7) in DAP response
Error: JTAG-DP STICKY ERROR
Error: Invalid ACK (7) in DAP response
Error: JTAG-DP STICKY ERROR
Error: Invalid ACK (7) in DAP response
Error: JTAG-DP STICKY ERROR
Error: Invalid ACK (7) in DAP response
Error: JTAG-DP STICKY ERROR
Error: Invalid ACK (7) in DAP response
Error: JTAG-DP STICKY ERROR
Error: Invalid ACK (7) in DAP response
Error: JTAG-DP STICKY ERROR
Error: Invalid ACK (7) in DAP response
Error: JTAG-DP STICKY ERROR
Error: Invalid ACK (7) in DAP response
Error: JTAG-DP STICKY ERROR

/home/francisco/workspace/RIOT/examples/hello-world/../../Makefile.include:669: recipe for target 'flash' failed
make: *** [flash] Error 1

```
</details>

<details><summary>master</summary>

```
BOARD=iotlab-m3 make -C examples/hello-world/ flash-only
/home/francisco/workspace/RIOT/dist/tools/openocd/openocd.sh flash /home/francisco/workspace/RIOT/examples/hello-world/bin/iotlab-m3/hello-world.elf
### Flashing Target ###
Open On-Chip Debugger 0.10.0+dev-00703-g92bb76a4-dirty (2019-07-19-14:27)
Licensed under GNU GPL v2
For bug reports, read
	http://openocd.org/doc/doxygen/bugs.html
Warn : Interface already configured, ignoring
Info : auto-selecting first available session transport "jtag". To override use 'transport select <transport>'.
adapter speed: 1000 kHz
adapter_nsrst_delay: 100
jtag_ntrst_delay: 100
none separate
cortex_m reset_config sysresetreq
trst_and_srst separate srst_nogate trst_push_pull srst_open_drain connect_deassert_srst
trst_and_srst separate srst_nogate trst_push_pull srst_open_drain connect_assert_srst
Error: libusb_get_string_descriptor_ascii() failed with LIBUSB_ERROR_INVALID_PARAM
Info : clock speed 1000 kHz
Info : JTAG tap: stm32f1x.cpu tap/device found: 0x3ba00477 (mfg: 0x23b (ARM Ltd.), part: 0xba00, ver: 0x3)
Info : JTAG tap: stm32f1x.bs tap/device found: 0x06414041 (mfg: 0x020 (STMicroelectronics), part: 0x6414, ver: 0x0)
Info : stm32f1x.cpu: hardware has 6 breakpoints, 4 watchpoints
Info : stm32f1x.cpu: external reset detected
Info : Listening on port 36127 for gdb connections
    TargetName         Type       Endian TapName            State       
--  ------------------ ---------- ------ ------------------ ------------
 0* stm32f1x.cpu       cortex_m   little stm32f1x.cpu       reset
Info : JTAG tap: stm32f1x.cpu tap/device found: 0x3ba00477 (mfg: 0x23b (ARM Ltd.), part: 0xba00, ver: 0x3)
Info : JTAG tap: stm32f1x.bs tap/device found: 0x06414041 (mfg: 0x020 (STMicroelectronics), part: 0x6414, ver: 0x0)
target halted due to debug-request, current mode: Thread 
xPSR: 0x01000000 pc: 0x0800045c msp: 0x20000200
auto erase enabled
Info : device id = 0x10016414
Info : flash size = 512kbytes
wrote 10240 bytes from file /home/francisco/workspace/RIOT/examples/hello-world/bin/iotlab-m3/hello-world.elf in 0.466561s (21.433 KiB/s)
verified 8540 bytes in 0.152391s (54.727 KiB/s)
Info : JTAG tap: stm32f1x.cpu tap/device found: 0x3ba00477 (mfg: 0x23b (ARM Ltd.), part: 0xba00, ver: 0x3)
Info : JTAG tap: stm32f1x.bs tap/device found: 0x06414041 (mfg: 0x020 (STMicroelectronics), part: 0x6414, ver: 0x0)
shutdown command invoked
Done flashing
```
</details>

### CODE REVIEW

File: makefiles/tools/openocd-adapters/iotlab.inc.mk [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# iotlab-usb debug adapter
+OPENOCD_ADAPTER_INIT ?= -c 'source [find interface/ftdi/iotlab-usb.cfg]'
```
Comment (aabadie) [inline makefiles/tools/openocd-adapters/iotlab.inc.mk]: This configuration file is already included by the openocd.cfg file of the iotlab-m3 board. But maybe it removed from the openocd.cfg file ?

Comment (fjmolinas) [reply makefiles/tools/openocd-adapters/iotlab.inc.mk]: They are used for different things, and `OPENOCD_ADAPTER_INIT` is evaluated before loading the configuration file if you look at `openocd.sh`.

**Comment by fjmolinas** [comment]
@aabadie I applied your changes request.

File: boards/iotlab-a8-m3/Makefile.include [hunk 2]
```diff
@@ -1,3 +1,7 @@
 USEMODULE += boards_common_iotlab
 
+DEBUG_ADAPTER ?= iotlab
```
Comment (aabadie) [inline boards/iotlab-a8-m3/Makefile.include]: Those 2 lines could be moved in the `common/iotlab/Makefile.include` file, same for the iotlab-la8-m3.

Comment (fjmolinas) [reply boards/iotlab-a8-m3/Makefile.include]: done

**Comment by fjmolinas** [comment]
@aabadie is it ok to squash?

**Comment by aabadie** [comment]
>  is it ok to squash?

Please go ahead!

**Comment by fjmolinas** [comment]
@aabadie all green here!

**Comment by fjmolinas** [comment]
Thanks for the  review

**Comment by miri64** [comment]
How do you get this ID? Maybe it makes sense to add this to the `list-ttys` make target.

**Comment by fjmolinas** [comment]
> How do you get this ID? Maybe it makes sense to add this to the list-ttys make target.

There is no ID by default on a M3, but you can write one to the ftdi eeprom as descriped in the PR. If that is done the serial shows up already;

```
/sys/bus/usb/devices/4-1.5.6: IoT-LAB M3, serial: 'TRIBE4441125307150001', tty(s): ttyUSB3, ttyUSB2
```

**Comment by miri64** [comment]
Ahh thanks! Sorry for the noise and obviously being lazy and not reading what is written ;-).

**Comment by miri64** [comment]
> `ftdi_eeprom --flash-eeprom ~/workspace/ci-riot-tribe/udev/iotlab-m3.conf`
> 
> ```
> 
> ```

Where do I get this file?

**Comment by fjmolinas** [comment]
> Where do I get this file?

I generated a configuration file from the configurations in https://github.com/iot-lab/iot-lab-ftdi-utils/blob/master/eeprom_config.c and added the necessary stuff to change and use the serial, the file is here https://github.com/fjmolinas/ci-riot-tribe/blob/master/udev/iotlab-m3.conf

**Comment by miri64** [comment]
Thanks!

**Pull request merged**